Sources/TuistGenerator/Mappers/ResourcesProjectMapper.swift

```diff
@@ -57,54 +57,78 @@ public class ResourcesProjectMapper: ProjectMapping {
-                    Bundle.main.resourceURL,
-
-                    // Bundle should be present here when the package is linked into a framework.
-                    Bundle(for: BundleFinder.self).resourceURL,
-
-                    // For command-line tools.
-                    Bundle.main.bundleURL,
-                ]
-
-                for candidate in candidates {
-                    let bundlePath = candidate?.appendingPathComponent(bundleName + ".bundle")
-                    if let bundle = bundlePath.flatMap(Bundle.init(url:)) {
-                        return bundle
+    static func fileContent(targetName: String, bundleName: String, target: Target) -> String {
+        if !target.supportsResources {
+            return """
+            // swiftlint:disable all
+            import Foundation
+
+            // MARK: - Swift Bundle Accessor
+
+            private class BundleFinder {}
+
+            extension Foundation.Bundle {
+                /// Since \(targetName) is a library, the bundle containing the resources is copied into the final product (app).
+                static var module: Bundle = {
+                    let bundleName = "\(bundleName)"
+
+                    let candidates = [
+                        Bundle.main.resourceURL,
+                        Bundle(for: BundleFinder.self).resourceURL,
+                        Bundle.main.bundleURL,
+                    ]
+
+                    for candidate in candidates {
+                        let bundlePath = candidate?.appendingPathComponent(bundleName + ".bundle")
+                        if let bundle = bundlePath.flatMap(Bundle.init(url:)) {
+                            return bundle
+                        }
                     }
-                }
-                fatalError("unable to find bundle named \(bundleName)")
-            }()
-        }
-
-        // MARK: - Objective-C Bundle Accessor
-
-        @objc
-        public class \(targetName.camelized.uppercasingFirst)Resources: NSObject {
-           @objc public class var bundle: Bundle {
-                 return .module
-           }
+                    fatalError("unable to find bundle named \(bundleName)")
+                }()
+            }
+
+            // MARK: - Objective-C Bundle Accessor
+
+            @objc
+            public class \(targetName.camelized.uppercasingFirst)Resources: NSObject {
+               @objc public class var bundle: Bundle {
+                     return .module
+               }
+            }
+            // swiftlint:enable all
+            """
+        } else {
+            return """
+            // swiftlint:disable all
+            import Foundation
+
+            // MARK: - Swift Bundle Accessor
+
+            private class BundleFinder {}
+
+            extension Foundation.Bundle {
+                /// Since \(targetName) is a framework, the bundle represents a framework.
```

Comment:
minor: `targetName` could be an app too, we could use something along the lines of:

> Since `\(targetName)` is a `\(target.product)`, the bundle for classes within this module can be used directly.